packages/components/src/progress-bar/stories/index.story.tsx

```diff
@@ -29,3 +29,34 @@ const Template: StoryFn< typeof ProgressBar > = ( { ...args } ) => {
 
 export const Default: StoryFn< typeof ProgressBar > = Template.bind( {} );
 Default.args = {};
+
+const withCustomWidthCustomCSS = `
+	.custom-progress-bar {
+		width: 100%;
+	}
+`;
+
+/**
+ * A progress bar with a custom overriden width.
+ *
+ * You can override the default `width` or further customize the behavior
+ * and visuals by passing a custom CSS class via the `cssName` prop.
+ *
+ * This example shows a progress bar with an overriden `width` of `100%` which
+ * makes it fit all available horizontal space of the parent element. The CSS
+ * class looks like this:
+ *
+ * ```css
+ * .custom-progress-bar {
+ *   width: 100%;
+ * }
+ * ```
+ */
+export const WithCustomWidth: StoryFn< typeof ProgressBar > = () => {
+	return (
+		<>
+			<style>{ withCustomWidthCustomCSS }</style>
+			<ProgressBar className="custom-progress-bar" />
+		</>
+	);
+};
```

Comment:
Here is a cleaner way we could do this, which keeps the autogenerated code snippet clean:

```suggestion
export const WithCustomWidth = Template.bind( {} );
WithCustomWidth.args = {
	className: 'custom-progress-bar',
};
WithCustomWidth.decorators = [
	( Story ) => (
		<>
			<style>{ withCustomWidthCustomCSS }</style>
			<Story />
		</>
	),
];
```

Another thing we need to be careful about is to make sure all the controls still work when adding a new story. So we can't just do `const NewStory = () => (<Foo />)`, but instead need to pass down props like `const NewStory = (props) => (<Foo {...props} />)`.

Reply:
Ah, true, I had done that in past iterations but forgot in this one. Thanks for catching that!